### ISSUE TITLE
java.lang.RuntimeException: Canvas: trying to use a recycled bitmap android.graphics.Bitmap@42e9b978

### ISSUE THREAD

**Issue by sirbralex** [moagrius/TileView#221]
Sometimes i get that error:

E/AndroidRuntime: java.lang.RuntimeException: Canvas: trying to use a recycled bitmap android.graphics.Bitmap@42e9b978
E/AndroidRuntime:     at android.graphics.Canvas.throwIfCannotDraw(Canvas.java:1084)
E/AndroidRuntime:     at android.view.GLES20Canvas.drawBitmap(GLES20Canvas.java:811)
E/AndroidRuntime:     at com.qozix.tileview.tiles.Tile.draw(Tile.java:159)
E/AndroidRuntime:     at com.qozix.tileview.tiles.TileCanvasView.drawTiles(TileCanvasView.java:68)
E/AndroidRuntime:     at com.qozix.tileview.tiles.TileCanvasView.onDraw(TileCanvasView.java:105)
E/AndroidRuntime:     at android.view.View.draw(View.java:14465)

**Comment by moagrius** [comment]
I have not seen that error and can't reproduce.  Can you provide the steps necessary to reproduce?

Also, for now, in https://github.com/moagrius/TileView/blob/master/tileview/src/main/java/com/qozix/tileview/tiles/Tile.java, try changing [line 133](https://github.com/moagrius/TileView/blob/master/tileview/src/main/java/com/qozix/tileview/tiles/Tile.java#L133) and [158](https://github.com/moagrius/TileView/blob/master/tileview/src/main/java/com/qozix/tileview/tiles/Tile.java#L158) to :

```
if( mBitmap != null && !mBitmap.isRecycled() )
```

**Comment by sierbrat** [comment]
I don't know how it happens. Sometimes during zoom.

**Comment by moagrius** [comment]
i'm adding a method to disable bitmap recycling, will be up shortly

**Comment by moagrius** [comment]
2.0.6 has a method to prevent bitmap recycling, `TileView.setShouldRecyleBitmaps`, in your case, you'd want to pass `false`

**Comment by moagrius** [comment]
did this method work for you, or are you still experiencing errors?

**Comment by azizbekian** [comment]
This issue persists in `2.2.3`. Happens when zooming out.

**Comment by moagrius** [comment]
will fix

**Comment by azizbekian** [comment]
I would like to describe the situation when this happens. Possibly I have a bad (customized) image getter, which sometimes may return "broken" bitmaps. When substituted it with authentic Picasso no exception occurs. Nevertheless, the lib should be ready to handle recycled bitmaps too.

**Comment by moagrius** [comment]
it's rare but i'm pretty sure i understand why it happens, previously i had a method to turn of recycling, but since that's a fairly old api anyway, i removed it.  it's fairly trivial to add it back in, which is why i put this issue in `willfix`

**Comment by moagrius** [comment]
fixed in 2.2.5 https://github.com/moagrius/TileView/releases/tag/2.2.5

```
compile 'com.qozix:tileview:2.2.5'
```
